draft-ietf-quic-transport.md

```diff
@@ -2298,10 +2305,12 @@ transport parameter in the TLS handshake.
 Servers MAY communicate a preferred address of each address family (IPv4 and
 IPv6) to allow clients to pick the one most suited to their network attachment.
 
-Once the handshake is finished, the client SHOULD select one of the two
+Once the handshake is confirmed, the client SHOULD select one of the two
 server's preferred addresses and initiate path validation (see
 {{migrate-validate}}) of that address using the connection ID provided in the
-preferred_address transport parameter.
+preferred_address transport parameter or, if that connection ID has been
+retired, any active connection ID that has not already been used on a different
+path.
```

Comment:
I can't suggest here, but maybe "using any previously unused connection ID, taken from either the preferred_address transport parameter or the NEW_CONNECTION_ID frame."

Reply:
I prefer @martinthomson's suggestion. For CIDs provided by NCID frames, we do not require them to be used in particular order, even when the client has received CIDs in order. I would be hesitant to to require in-order use of CID when SPA is involved.

Reply:
Thanks! I was struggling to find a way to word that nicely, this is much better.

Reply:
Fixed in 5509f1d